src/conversation_engine/engine.py

```diff
@@ -1,36 +1,40 @@
 # System
+import json
 import os
+import random
 from copy import deepcopy
+import yaml
 
 # ROS
-import actionlib
-import random
 import rospy
-import json
-import yaml
+from std_msgs.msg import String
 
 # TU/e Robotics
 from action_server import Client, TaskOutcome
 from grammar_parser import cfgparser
-from std_msgs.msg import String
 
-# Conversation engine
-from conversation_engine.msg import ConverseAction, ConverseResult, ConverseFeedback
 
 def sanitize_text(txt):
     stripped = "".join(c for c in txt if c not in """!.,:'?`~@#$%^&*()+=-/\></*-+""")
     lowered = stripped.lower()
 
-    mapping = { "dining table": "dining_table",
-                "display case": "display_case",
-                "storage shelf": "storage_shelf",
-                "couch table": "couch_table",
-                "tv table": "tv_table",
-                "kitchen table": "kitchen_table",
-                "kitchen cabinet": "kitchen_cabinet",
-                "side table": "side_table",
-                "living room": "living_room",
-                "dining room": "dining_room"}
+    mapping = {"dining table": "dining_table",
+               "display case": "display_case",
+               "storage shelf": "storage_shelf",
+               "couch table": "couch_table",
+               "tv table": "tv_table",
+               "kitchen table": "kitchen_table",
+               "kitchen cabinet": "kitchen_cabinet",
+               "side table": "side_table",
+               "living room": "living_room",
+               "dining room": "dining_room",
+               "storage table":"storage_table",
+               "end table":"end_table",
+               "chocolate drink":"chocolate_drink",
+               "grape juice":"grape_juice",
+               "orange juice":"orange_juice",
+               "potato chips":"potato_chips",
+               "cleaning stuff":"cleaning_stuff"}
```

Comment:
All of this should be in the knowledge, I guess. Indeed, the required information is already in the knowledge. You could just get all object types from the knowledge, replace the underscores with spaces and use those as keys to the original object types. You will get the same dictionary as the one you're creating manually here.

Reply:
I think a bunch of these can be auto-generated as well. And some fuzzy matching could help as well

Reply:
Is there an issue for this yet? If there is, we can merge.

Reply:
Issue created. That is #19